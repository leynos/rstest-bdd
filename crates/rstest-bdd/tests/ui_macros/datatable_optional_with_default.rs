//! Compile-fail fixture: optional fields cannot also specify defaults.

use rstest_bdd_macros::DataTableRow;

#[derive(Debug, DataTableRow)]
struct InvalidOptionalDefault {
    #[datatable(optional, default)]
    value: Option<String>,
}

fn main() {}
