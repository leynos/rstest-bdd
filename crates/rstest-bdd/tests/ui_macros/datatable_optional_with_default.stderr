error: optional fields cannot specify a default
 --> tests/ui_macros/datatable_optional_with_default.rs:7:5
  |
7 | /     #[datatable(optional, default)]
8 | |     value: Option<String>,
  | |_________________________^
